### PR TITLE
docs: Document the retain feature

### PR DESCRIPTION
**Pull request by mhan83**
### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

File: docs/web-apps/automated-testing/_partials/_advanced.md [hunk 1]
```diff
@@ -153,12 +153,36 @@ This feature is highly experimental.
...
+  retain:
+    report-directory: archived-report.zip
+
+env:
+  SAUCE_SYNC_WEB_ASSETS: "true"
+```
+
+When configured this way, the directory named `report-directory` will be archived as `archived-report.zip`.
+To maintain backwards compatibility, some asset types (e.g. images, logs, etc.) in `report-directory` will be
+automatically attached to the test results.
```
Comment (alexplischke) [inline docs/web-apps/automated-testing/_partials/_advanced.md:185]: Hmm. How about:
```suggestion
To maintain backwards compatibility with our UI, some asset types (e.g. images, logs, etc.) in `report-directory` will still be automatically copied over to `__assets__/` and attached to the test results.
```

File: docs/web-apps/automated-testing/_partials/_advanced.md [hunk 2]
```diff
@@ -153,12 +153,36 @@ This feature is highly experimental.
...
+in your saucectl config to opt out of the default behaviour. When set, the configured output directory
+for the test run will be honoured; e.g. it won't be overridden to `__assets__/`.
+
+```yaml title= "example configuration"
+artifacts:
+  retain:
+    report-directory: archived-report.zip
+
+env:
+  SAUCE_SYNC_WEB_ASSETS: "true"
```
[the remark this answers is not in the document]
Comment (tianfeng92) [reply docs/web-apps/automated-testing/_partials/_advanced.md:180]: oops... Clicked the approve button 😂

File: docs/web-apps/automated-testing/cucumberjs-playwright/yaml.md [hunk 3]
```diff
@@ -587,6 +587,35 @@ artifacts:
...
+    source-directory: destination-archive.zip
+  download:
+    when: always
+    match:
+      - destination-archive.zip
+    directory: ./artifacts/
+```
+
+:::note
+The source and destination will be relative to the `rootDir` defined in your configuration.
```
[the remark this answers is not in the document]
Comment (mhan83) [reply docs/web-apps/automated-testing/cucumberjs-playwright/yaml.md:610]: Yah, I was waffling on this. I thought this was the most succinct way to describe it while being close enough to the truth.

File: docs/web-apps/automated-testing/_partials/_advanced.md [hunk 4]
```diff
@@ -153,12 +153,35 @@ This feature is highly experimental.
...
+you can use the `retain` feature to define a directory to archive and store as a test asset.
+
+### Handling nested assets
 
-:::note Nested Paths
 Nested assets are stored **flat** in Sauce Labs. A test asset like `__assets__/mylogs/log.txt` would therefore be stored and available for download as `log.txt`.
 Please keep that in mind when creating custom assets, as examples like `__assets__/mylogs/log.txt` and `__assets__/myotherlogs/log.txt` would eventually collide when persisted.
-:::
+
+For cypress and playwright, there are cases where you may want to override this default behaviour; e.g. your test framework generates
```
[the remark this answers is not in the document]
Comment (mhan83) [reply docs/web-apps/automated-testing/_partials/_advanced.md:168]: :us:

**Review by alexplischke — APPROVED** [review]

**Pull request merged**